src/store/entities/constants.ts

```diff
@@ -1 +1,28 @@
 export const SAVE_NORMALIZED_ENTITIES = 'SAVE_NORMALIZED_ENTITIES';
+
+export const TYPE_DOMAIN_MAPPING = {
```

Comment:
Using this map seems quite hard to maintain. New entities will always have to be added before you can use them. And what about using it with internal(/undocumented) endpoints? This will not work (and even error I think?) when an endpoint returns an entity that's not in this list.

Is there a dynamic alternative that we can use? I wouldn't even care that it's grammatically incorrect (e.g. `companys`).

Reply:
Hmm, not sure, I don't really see a different way to transition from the singular entity to domain version. This could be solved if the sideloading reference just returned `type: companies`, that way we wouldn't need the reverse mapping in `api-gateway` either

The only thing that comes to mind is maybe having a fallback to the simple `+ s` version?